The initial committed stack size is insufficient for Rails and Bundler:

http://cygwin.com/ml/cygwin/2013-02/msg00109.html
http://stackoverflow.com/a/14790573

The -Xlinker form must be used because with the -Wl, form, commas are used
to separate arguments when passed to ld(1), and here the comma is part of
the argument (--stack reserve[,commit]).

--- origsrc/ruby-1.9.3-p385/configure.in	2012-11-13 04:02:45.000000000 -0600
+++ src/ruby-1.9.3-p385/configure.in	2013-02-11 04:17:10.036726700 -0600
@@ -2101,7 +2101,7 @@ if test "$with_dln_a_out" != yes; then
 			rb_cv_dlopen=yes],
 	[cygwin*|mingw*], [
 			: ${LDSHARED='$(CC) -shared $(if $(filter-out -g -g0,$(debugflags)),,-s)'}
-			XLDFLAGS="$XLDFLAGS -Wl,--stack,0x00200000,--enable-auto-import"
+			XLDFLAGS="$XLDFLAGS -Xlinker --stack -Xlinker 0x00200000,0x00040000 -Xlinker --enable-auto-import"
 			DLDFLAGS="${DLDFLAGS} -Wl,--enable-auto-image-base,--enable-auto-import"
 			: ${LIBPATHENV=""}
 			rb_cv_dlopen=yes],
